### PR TITLE
Adapt drawer layout for smaller screens

### PR DESCRIPTION
**Pull request by msmanoj**
## Description
Updated the drawer layout to adapt to smaller screens according to the XD.

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1570

## Deployment URL(s)
http://gb-drawer-layout.review.ensembl.org

## Views affected
Genome browser -> Drawer

### CODE REVIEW

**Comment by imransl** [comment]
There is now a right hand border in the track panel along with white space. Probably due to reducing the width of the side wrapper?

Screenshot 2022-04-25 at 17 11 00

**Comment by azangru** [comment]
Not sure I understand the changes in this PR.

Here's a [quick mockup](https://jsfiddle.net/ruyq6cd2) of what I thought would be the change. Is my model wrong?

**Comment by msmanoj** [comment]
> There is now a right hand border in the track panel along with white space. Probably due to reducing the width of the side wrapper?
> 
> Screenshot 2022-04-25 at 17 11 00
> 
> Also the position of the close button in the drawer does not match that of the XD.
> 
> XD: Screenshot 2022-04-25 at 17 21 56
> 
> Review site: Screenshot 2022-04-25 at 17 21 44

@imransl - these should now be fixed

File: src/content/app/genome-browser/components/drawer/drawer-views/transcript-summary/TranscriptSummary.scss [hunk 1]
```diff
@@ -84,6 +75,7 @@
   align-items: center;
   column-gap: 30px;
   row-gap: 10px;
+  width: 100%;
```
Comment (azangru) [inline src/content/app/genome-browser/components/drawer/drawer-views/transcript-summary/TranscriptSummary.scss]: isn't this what `display: flex` does by default?

File: src/content/app/genome-browser/components/drawer/drawer-views/transcript-summary/TranscriptSummary.scss [hunk 2]
```diff
@@ -21,33 +21,24 @@
...
       margin-left: 10px;
     }
 
     & > div + div {
       margin-top: 3px;
     }
+
+    &.responsive {
+      width: calc(100vw - 320px - 45px - 46px - 220px);
+    }
```
Comment (azangru) [inline src/content/app/genome-browser/components/drawer/drawer-views/transcript-summary/TranscriptSummary.scss]: Hmm, this looks suspicious.

File: src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+$drawerWindowWidth: 45px;
+$sidebarWithToolstripWidth: calc(320px + 46px);
+$summaryLabelWidth: 130px;
+$summaryLabelWithGap: calc(#{$summaryLabelWidth} + 10px);
+$summaryPadding: 56px;
+//  drawerWindowWidth (45px)     summaryPadding left (30px)              summaryPadding right(26px)
+//    |                                |                                              |
+// [----][-------------------------][-----][------------][************************][-----]
+//              |                                |                    |
+//    sidebarWithToolstripWidth (366px)    summaryLabelWithGap   summaryContentWidth
```
Comment (azangru) [inline src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss:12]:
```
/*
multi
line
comment
*/
```
?

Comment (msmanoj) [reply src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss:12]: VSCode seems to do some clever stuff to confuse us if I change it to multi line comment :sweat_smile:. Is it still worth changing?
Screenshot 2022-04-28 at 09 16 06

File: src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
   .value {
     grid-column: value;
-
+    min-width: 880px;
```
Comment (azangru) [inline src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss:18]: this seems unnecessary?

File: src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss [hunk 5]
```diff
@@ -36,6 +36,13 @@
   }
 }
 
+.responsiveRow {
+  > .value {
+    width: 100%;
+    min-width: 100%;
```
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/content/app/genome-browser/components/drawer/drawer-views/gene-summary/GeneSummary.scss]: I don't know how I ended up with adding a `min-width` when it doesn't make any difference other than showing an unwanted scrollbar. 

Removing the `width` & `min-width` works fine. I think it's the power of `Display: flex`?

**Comment by azangru** [comment]
For the drawer:

close button: 14px right, 15px top
right padding: 30px

**Review by azangru — APPROVED** [review]
We could also probably improve the min-height of the sequence component; but can do so in a separate PR.

**Pull request merged**